File: Leetcode_solutions/add-binary.js
/**
 * @param {string} a
 * @param {string} b
 * @return {string}
 */
var addBinary = function(a, b) {
  _a = a.length - 1;
  _b = b.length - 1;
  carry = 0;
  sum = [];
  digit = _a + _b + 2;
  while (_a >= 0 || _b >= 0) {
    current = parseInt(a[_a--] || 0) + parseInt(b[_b--] || 0) + carry;
    newDigit = current % 2;

    carry = Math.floor(current / 2);
    sum[digit--] = newDigit;
  }
  if (carry) {
    sum.splice(0, 0, carry);
  }
  return sum.join("");
};

File: Leetcode_solutions/Add Binary.py
class Solution:
    def addBinary(self, a: str, b: str) -> str:
        i=0
        ret=""
        carry=0
        while i<max(len(a),len(b)) or carry>0:
            aC=0
            bC=0
            if i<len(a):
                aC=int(a[len(a)-i-1])
            if i<len(b):
                bC=int(b[len(b)-i-1])
            ret = str(aC ^ bC ^ carry)+ret
            carry = ((aC ^ bC) & carry) | (aC & bC)
            i += 1
        return ret